stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/client/WebSocketClientHandshakeListener.java

```diff
@@ -84,17 +108,84 @@ public void onError(Throwable throwable, HttpCarbonResponse response) {
+                                                                         ObjectValue webSocketConnector,
+                                                                         ObjectValue webSocketClient,
+                                                                         WebSocketService wsService) {
         WebSocketConnectionInfo connectionInfo = new WebSocketConnectionInfo(
                 wsService, webSocketConnection, webSocketClient);
         webSocketConnector.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_CONNECTION_INFO, connectionInfo);
         webSocketClient.set(WebSocketConstants.CLIENT_CONNECTOR_FIELD, webSocketConnector);
         return connectionInfo;
     }
+
+    private static void setWebSocketClient(ObjectValue webSocketClient, HttpCarbonResponse carbonResponse,
+                                           WebSocketConnection webSocketConnection, RetryContext retryConfig) {
+        //Using only one service endpoint in the client as there can be only one connection.
+        webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD, HttpUtil.createResponseStruct(carbonResponse));
+        if (retryConfig != null && retryConfig.isFirstConnectionMadeSuccessfully()) {
+            webSocketClient.set(WebSocketConstants.LISTENER_ID_FIELD, webSocketConnection.getChannelId());
+        } else {
+            WebSocketUtil.populateWebSocketEndpoint(webSocketConnection, webSocketClient);
+        }
+    }
+
+    /**
+     * Call the readNextFrame().
+     *
+     * @param readyOnConnect ready on connect
+     * @param webSocketClient webSocket client
+     * @param webSocketConnection webSocket connection
+     */
+    private static void readNextFrame(boolean readyOnConnect, ObjectValue webSocketClient,
+                                      WebSocketConnection webSocketConnection, RetryContext retryConfig) {
+        if (readyOnConnect || ((boolean) (webSocketClient.get(WebSocketConstants.CONNECTOR_IS_READY_FIELD)))) {
+            if (retryConfig != null && !retryConfig.isFirstConnectionMadeSuccessfully()) {
+                WebSocketUtil.readFirstFrame(webSocketConnection, webSocketClient);
+            } else {
+                webSocketConnection.readNextFrame();
+            }
+        }
+    }
+
+    /**
+     * Count Down the initialised countDownLatch.
+     *
+     * @param webSocketClient webSocket client
+     */
+    private static void countDownForHandshake(ObjectValue webSocketClient) {
+        if (webSocketClient.getNativeData(WebSocketConstants.COUNT_DOWN_LATCH) != null) {
+            ((CountDownLatch) webSocketClient.getNativeData(WebSocketConstants.COUNT_DOWN_LATCH)).countDown();
+            webSocketClient.addNativeData(WebSocketConstants.COUNT_DOWN_LATCH, null);
+        }
+    }
+
+    /**
+     * Set the value into the retryContext.
+     *
+     * @param retryConfig retry context
```

Comment:
```suggestion
     * @param retryConfig retry context
```
Need more explanation.